utils.py

```diff
@@ -103,5 +104,22 @@ def get_sentence_frame_acc(intent_preds, intent_labels, slot_preds, slot_labels)
 
     sementic_acc = np.multiply(intent_result, slot_result).mean()
     return {
-        "sementic_frame_acc": sementic_acc
+        "semantic_frame_acc": sementic_acc
     }
+
+
+class ResultsLogger:
+
+    def __init__(self, csv_filename):
+        self.csv_file = open(csv_filename, 'w')
+        self.writer = None
+
+    def write_results(self, results):
+        if not self.writer:
+            self.writer = csv.DictWriter(self.csv_file, fieldnames=results.keys())
+            self.writer.writeheader()
+        self.writer.writerow(results)
+        self.csv_file.flush()
```

Comment:
So even if I stopped the training by killing it, the training stats would be saved in the csv file. Is that what `flush` does?

Reply:
Correct. The file is properly closed on the end of training but flush guarantees we do not lose the metrics data if the process is terminated prematurely.